tests/scripts/all.sh

```diff
@@ -810,7 +810,7 @@ make CC=gcc CFLAGS='-Werror -Os'
 
 
 if uname -a | grep -F Linux >/dev/null; then
-    msg "build/test: make shared" # ~ 40s
+    msg "build+test: make shared" # ~ 40s
```

Comment:
This is the only line changed by a commit whose message is “fix incorrect use of if_build_succeeded”. What gives?

Reply:
After rebasing, I reviewed each commit and either deleted or amended the ones that no longer made sense. Apparently I should have done a second pass and checked that the commit actually made sense after amending. Sorry for that.

Should I rewrite history to drop this commit, or is it too late now that the review process has started? Or maybe I could do it after review of the main branch is completed successfully (but before doing the backports) so that in the end people can just check that the diff after rewriting is only this line.

Reply:
My preference would be for you to meld this commit into another one after this PR has been reviewed. Save the original commit in a branch (my personal convention is `cleanup-all.sh-1`) so that Github doesn't delete the commit.